crates/holochain/src/conductor/conductor/tests/agent_key_revocation.rs

```diff
@@ -0,0 +1,908 @@
+        let bob_app = apps.next().unwrap();
+        let bob_cell_1 = bob_app.cells().get(0).unwrap();
+        let bob_cell_2 = bob_app.cells().get(1).unwrap();
+
+        // Await initial DHT sync of Dpki and both cells
+        await_consistency(
+            20,
+            [
+                &conductors[0].dpki_cell().unwrap(),
+                &conductors[1].dpki_cell().unwrap(),
+            ],
+        )
+        .await
+        .unwrap();
+        await_consistency(20, [alice_cell_1, bob_cell_1])
+            .await
+            .unwrap();
+        await_consistency(5, [alice_cell_2, bob_cell_2])
+            .await
+            .unwrap();
+
+        // Revoke agent key in Dpki
+        revoke_agent_key_in_dpki(&conductors[0], alice.clone()).await;
+
+        // Await for revocatino to reach bob's Dpki
+        await_consistency(
+            20,
+            [
+                &conductors[0].dpki_cell().unwrap(),
+                &conductors[1].dpki_cell().unwrap(),
+            ],
+        )
+        .await
+        .unwrap();
+
+        // Delete agent key of cell 1 of the app and publish and integrate ops
+        let alice_source_chain_1 = SourceChain::new(
+            conductors[0]
+                .get_or_create_authored_db(alice_cell_1.dna_hash(), alice.clone())
+                .unwrap(),
+            conductors[0].get_dht_db(alice_cell_1.dna_hash()).unwrap(),
+            conductors[0]
+                .get_dht_db_cache(alice_cell_1.dna_hash())
+                .unwrap(),
+            conductors[0].keystore().clone(),
+            alice.clone(),
+        )
+        .await
+        .unwrap();
+        alice_source_chain_1
+            .delete_valid_agent_pub_key()
+            .await
+            .unwrap();
+        alice_source_chain_1
+            .flush(&conductors[0].holochain_p2p().to_dna(
+                alice_cell_1.dna_hash().clone(),
+                conductors[0].get_chc(alice_cell_1.cell_id()),
+            ))
+            .await
+            .unwrap();
+        conductors[0]
+            .get_cell_triggers(alice_cell_1.cell_id())
+            .await
+            .unwrap()
+            .publish_dht_ops
+            .trigger(&"key deletion");
+        conductors[0]
+            .get_cell_triggers(alice_cell_1.cell_id())
+            .await
+            .unwrap()
+            .integrate_dht_ops
+            .trigger(&"key deletion");
+
+        // Check agent key is invalid in cell 1
+        let invalid_agent_key_error = alice_source_chain_1
+            .valid_create_agent_key_action()
+            .await
+            .unwrap_err();
+        assert_matches!(invalid_agent_key_error, SourceChainError::InvalidAgentKey(invalid_key, cell_id) if invalid_key == alice && cell_id == *alice_cell_1.cell_id());
+
+        // Wait for key deletion on source chain to sync with bob
+        await_consistency(20, [alice_cell_1, bob_cell_1])
+            .await
+            .unwrap();
```

Comment:
This is a PITA but I'm not sure that this will check that the Delete was integrated as valid. It could have been integrated as rejected.

I wonder if there's a way to check the block list, or if you could just run a quick query to show that the Delete is valid

Reply:
If you comment out this line consistency is not reached https://github.com/holochain/holochain/pull/3930/files#diff-ecca2cd75dc5730737f8aa8e6f7915a0221290c348272bb8563bb9ba38da5e0fR434

Reply:
But I can check for the Delete.

Reply:
the fact of the op having been gossiped and integrated (consistency) is a different matter than the validation status after integration

Reply:
Well it seems that consistency of deterministic validation should mean the same thing on all sides. What about changing `await_consistency` so that it takes validation status into consideration?

Reply:
@maackle

> the fact of the op having been gossiped and integrated (consistency) is a different matter than the validation status after integration

In this particular case it works out, because ops with a dependency are put back into limbo and are not integrated when rejected. The `Delete` has a dependency, so it's not integrated. That's why commenting out the line above makes the test fail.

But I appreciate you pointing this out, as I was under the assumption that consistency here meant equality. Whether that should be the case is another discussion.

I've added the check for a validated `Delete` Action of the agent key.